spock-core/src/main/java/org/spockframework/runtime/ParameterizedFeatureNode.java

```diff
@@ -30,27 +43,151 @@ public SpockExecutionContext execute(SpockExecutionContext context, DynamicTestE
     }
+
+    // do not try to aggregate iteration results if they are reported individually
+    if (featureInfo.isReportIterations()) {
+      return context;
+    }
+
+    if (testExecutor.results.containsKey(FAILED)) {
+      List<Throwable> failures = testExecutor.results
+        .get(FAILED)
+        .stream()
+        .filter(Objects::nonNull)
+        .collect(toList());
+
+      switch (failures.size()) {
+        case 0:
+          throw new SpockAssertionError("At least one iteration failed");
+
+        case 1:
+          sneakyThrow(failures.get(0));
+
+        default:
+          MultipleFailuresError multipleFailuresError = new MultipleFailuresError(null, failures);
+          multipleFailuresError.setStackTrace(failures.get(0).getStackTrace());
+          throw multipleFailuresError;
+      }
+    } else if (!testExecutor.results.containsKey(SUCCESSFUL)) {
+      List<Throwable> abortions = testExecutor.results
+        .get(ABORTED)
+        .stream()
+        .filter(Objects::nonNull)
+        .collect(toList());
+
+      Throwable abortion;
+      switch (abortions.size()) {
+        case 0:
+          abortion = new TestAbortedException("All iterations were aborted");
+          break;
+
+        case 1:
+          abortion = abortions.get(0);
+          break;
+
+        default:
+          abortion = new MultipleFailuresError(null, abortions);
+          abortion.setStackTrace(abortions.get(0).getStackTrace());
+          break;
+      }
+
+      TestAbortedException testAbortedException;
+      if (abortion instanceof TestAbortedException) {
+        testAbortedException = (TestAbortedException) abortion;
+      } else {
+        testAbortedException = new TestAbortedException("All iterations were aborted", abortion);
+        testAbortedException.setStackTrace(abortion.getStackTrace());
+      }
+      throw testAbortedException;
+    }
+
     return context;
   }
 
   @Override
   public Type getType() {
-    return Type.CONTAINER;
+    return featureInfo.isReportIterations() ? Type.CONTAINER_AND_TEST : Type.TEST;
   }
 
   class ChildExecutor implements DynamicTestExecutor {
 
     private AtomicInteger executionCounter = new AtomicInteger();
 
+    private Map<Status, List<Throwable>> results = new HashMap<>();
+
     private final DynamicTestExecutor delegate;
 
     ChildExecutor(DynamicTestExecutor delegate) {this.delegate = delegate;}
 
     @Override
     public void execute(TestDescriptor testDescriptor) {
       executionCounter.incrementAndGet();
-      addChild(testDescriptor);
-      delegate.execute(testDescriptor);
+      if (featureInfo.isReportIterations()) {
+        addChild(testDescriptor);
+        delegate.execute(testDescriptor);
+      } else {
+        addChild(testDescriptor);
+        //TODO: replace by result of https://github.com/junit-team/junit5/issues/2188
+        execute(delegate, testDescriptor, new EngineExecutionListener() {
```

Comment:
Reading the code one can see a private method with a lot of reflection and delegation, but it would be good to mention what it is all about.

Reply:
Where do you mean? Line comments in the work-around code? JavaDoc for the work-around method? I still hope this will not make it into master as is, even if it is better than having nothing :-D

Reply:
Here, in code. There is `if` with addChild and delegate and else with addChild, delegate is combined with listener defined inline. I would like to have here some explanation what is this workaround for, such as `executeWithoutIterationReporting()`.

Reply:
The `addChild` in the branches were a refactoring left-over, I pulled it out before the `if`.
Isn`t the `if (featureInfo.isReportIterations())` excatly what you are asking for?
Is this too unclear in your opinion?

Reply:
Long "hacky" private method `execute()` is unclear in my opinion. I needed to get through the `if` statement which is 20+ lines away from that place. In my opinion the proposed method name might be more clear than that another `execute()` in the same class.

Reply:
Well, `executeWithoutIterationReporting` is not really appropriate, it is just `executeWithOwnListener` and as the proposed new method is also just `execute` with an additional listener parameter, I adopted this here. But I renamed it to `executeReplacingListener` now and added some line comments.